code/game/turfs/space/space.dm

```diff
@@ -54,12 +54,12 @@
 				qdel(A)
 				return
 
-			var/specials_list = recursive_type_check(A, list(/obj/item/weapon/disk/nuclear, /mob/living/silicon/robot/mommi))
+			var/specials_list = search_contents_for(A, list(/obj/item/weapon/disk/nuclear, /mob/living/silicon/robot/mommi))

```

Comment:
It looks like the logic order is wrong.
The comments suggest the behaviour you're talking about, but the skip comes after the atom is initially added, and before its contents are checked.